src/Operations.jl

```diff
@@ -328,9 +328,11 @@ end
 # all versioned packges should have a `tree_hash`
 function resolve_versions!(ctx::Context, pkgs::Vector{PackageSpec})
     # compatibility
-    v = intersect(VERSION, project_compatibility(ctx, "julia"))
-    if isempty(v)
-        @warn "julia version requirement for project not satisfied" _module=nothing _file=nothing
+    if ctx.julia_version !== nothing
```

Comment:
I think this means that the julia compatibility of the current project is ignored when running with `julia_version=nothing`. I think this needs to be passed to the resolver so that the julia compatibility is respected.

Reply:
Can you explain why you think that is a bad idea?  The whole reasoning behind `julia_version = nothing` is for it to ignore all Julia version constraints during resolution.  This is what allows us to, for instance, resolve versions of JLLs that obey the compat constraints between stdlib JLLs, but ignoring the Julia version constraints.

Reply:
Okay I understand. I thought it would treat julia as a normal package (with a non-fixed version) and the resolver would give back the most recent julia version that satisfies all the julia constraints of the packages that end up in the environment. But for that, the set of available julia versions has to be available which it currently is not. If https://github.com/JuliaRegistries/General/blob/master/J/julia/Versions.toml is updated then this could be done.

Reply:
Hmmm, that’s interesting. In what situation would you want that?

Reply:
It's been discussed in the context of juliaup in the situation where you, for example, have a project file (but no manifest) and want to launch the most recent julia that also satisfies your resolved project. You would then "bootstrap" this by running some julia version that resolves the environment with the julia version "floating" and then after, find out what julia version was actually resolved to and launch that.

Reply:
Maybe we set julia_version to “missing”, heh.